### PR TITLE
libgeotiff: fix builddirs + add module file to cmake find like generators only

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
All recipes which override builddirs (default value is base path of package folder) must be fixed, it can break downstream relying on cmake commands like `find_library()`

File: recipes/libgeotiff/all/conanfile.py [hunk 1]
```diff
@@ -66,39 +69,59 @@ def package(self):
...
+    @property
+    def _module_target_file(self):
         return "conan-official-{}-targets.cmake".format(self.name)
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "GeoTIFF"
         self.cpp_info.names["cmake_find_package_multi"] = "geotiff"
-        self.cpp_info.builddirs = [self._module_subfolder]
-        self.cpp_info.build_modules = [os.path.join(self._module_subfolder, self._module_file)]
+        self.cpp_info.builddirs.append(self._module_subfolder)
```
Comment (prince-chrismc) [inline recipes/libgeotiff/all/conanfile.py:122]: Think aloud here... There must be a technical detail about how this works which I am not familiar with.

If I understand correctly the previous way by assigning just this subfolder meant it **only looked** in `package_folder/lib/cmake` for the `FindPackage.cmake` which causes things to break.

The default is `[""]` which Conan internal translates to the `package_folder` (🧙) 

❓ If we put nothing in that folder why does it matter? What is this settings overriding the causes the CMake to look in the wrong path(s)?

The generated files are are output into the build folder which I presume must be included?

***

The docs https://docs.conan.io/en/latest/reference/conanfile/attributes.html#cpp-info which do not help me understand

NAME | DESCRIPTION
-- | --
`self.cpp_info.builddirs` | Ordered list with build scripts directory paths. Defaulted to [""] (Package folder directory)CMake generators will search in these dirs for files like findXXX.cmake

File: recipes/libgeotiff/all/conanfile.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
+from conans import ConanFile, CMake, tools
 import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
```
Comment (prince-chrismc) [inline recipes/libgeotiff/all/conanfile.py:5]: What new feature are you using?

File: recipes/libgeotiff/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:122]: Explained above: https://github.com/conan-io/conan-center-index/pull/4707#issuecomment-787102357

File: recipes/libgeotiff/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:5]: build_modules assignement per generator

File: recipes/libgeotiff/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:122]: `builddirs` seems to have more responsability than the doc says, e.g setting `CMAKE_PREFIX_PATH` in `cmake` generator

**Review by prince-chrismc — APPROVED** [review]
All the others of these were merged

**Review by uilianries — APPROVED** [review]

**Review by mathbunnyru — APPROVED** [review]

**Pull request merged**